File: 00_ETC_PY/29163.py
n = int(input())
v = list(map(int, input().split()))

holCnt = 0
jjakCnt = 0

for i in range(n):
    if v[i] % 2 == 0:
        jjakCnt += 1
    else:
        holCnt += 1

if jjakCnt > holCnt:
    print("Happy")
else:
    print("Sad")

File: 00_ETC/29163.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    cin.tie(0)->sync_with_stdio(0);

    int n;
    cin >> n;
    vector<int> v(n);

    for (int i = 0; i < n; ++i) {
        cin >> v[i];
    }
    int jjakCnt = 0, holCnt = 0;
    for (int i = 0; i < n; ++i) {
        if (v[i] % 2 != 0) {
            ++holCnt;
        } else {
            ++jjakCnt;
        }
    }
    if (jjakCnt > holCnt) {
        cout << "Happy";
    } else {
        cout << "Sad";
    }
    return 0;
}
//
// Created by 최승우 on 25. 10. 30.
//
